s2spy/calendar_shifter.py

```diff
@@ -0,0 +1,169 @@
+import copy
+from typing import Dict
+from typing import List
+from typing import Union
+import xarray as xr
+import s2spy.time
+
+
+def _gap_shift(
+    interval: s2spy.time.Interval, shift: Union[str, Dict[str, int]]
+) -> Dict[str, int]:
+    """
+    Shift a gap from a calendar interval by an input of a pandas-like
+    frequency string (e.g. "10d", "2W", or "3M"), or a pandas.DateOffset
+    compatible dictionary such as {days=10}, {weeks=2}, or {months=1, weeks=2}..
+    Args:
+        gap: a pandas DateOffset from a calendar instance.
+        shift: the shift for the gap
+    Returns:
+        dict with gap offset by shift
+    """
+    # pylint: disable=protected-access
+    if isinstance(interval.gap, str):
+        gap_time_dict = interval._parse_timestring(interval.gap)
+    else:
+        gap_time_dict = interval.gap
+    # pylint: disable=protected-access
+    if isinstance(shift, str):
+        shift_time_dict = interval._parse_timestring(shift)
+    else:
+        shift_time_dict = shift
+    # make the shift negative for the precursor to shift forward in time
+    if interval.role == "precursor":
+        shift_time_dict.update(
+            (key, value * -1) for key, value in shift_time_dict.items()
+        )
+    # combine gap and shift time dict
+    gap_shifted = {
+        k: gap_time_dict.get(k, 0) + shift_time_dict.get(k, 0)
+        for k in set(gap_time_dict) | set(shift_time_dict)
+    }
+
+    return gap_shifted
```

Comment:
```suggestion
    return {
        k: gap_time_dict.get(k, 0) + shift_time_dict.get(k, 0)
        for k in set(gap_time_dict) | set(shift_time_dict)
    }
```
You immediately return it anyway, no need for the `gap_shifted` variable.